diff -Naru mrbs.OLD/day.php mrbs/day.php
--- mrbs.OLD/day.php	2018-02-11 11:49:59.000000000 +0100
+++ mrbs/day.php	2018-10-22 09:31:50.091046221 +0200
@@ -51,25 +51,13 @@
 //t? are year, month and day of tomorrow
 
 // find the last non-hidden day
-$d = $day;
-do
-{  
-  $d--;
-  $i= mktime(12,0,0,$month,$d,$year);
-}
-while (is_hidden_day(date("w", $i)) && ($d > $day - 7));  // break the loop if all days are hidden
+$i = get_day($month, $day, $year, -1);
 $yy = date("Y",$i);
 $ym = date("m",$i);
 $yd = date("d",$i);
 
 // find the next non-hidden day
-$d = $day;
-do
-{
-  $d++;
-  $i= mktime(12, 0, 0, $month, $d, $year);
-}
-while (is_hidden_day(date("w", $i)) && ($d < $day + 7));  // break the loop if all days are hidden
+$i = get_day($month, $day, $year, +1);
 $ty = date("Y",$i);
 $tm = date("m",$i);
 $td = date("d",$i);
diff -Naru mrbs.OLD/edit_entry.php mrbs/edit_entry.php
--- mrbs.OLD/edit_entry.php	2018-02-11 11:49:59.000000000 +0100
+++ mrbs/edit_entry.php	2018-10-17 16:57:06.000000000 +0200
@@ -467,7 +467,7 @@
 
 function create_field_entry_type($disabled=FALSE)
 {
-  global $booking_types, $type, $is_mandatory_field;
+  global $booking_types, $type, $is_mandatory_field, $is_admin, $restricted_booking_types;
   
   // Don't bother with types if there's only one of them (or even none)
   if (count($booking_types) < 2)
@@ -493,7 +493,8 @@
   
   foreach ($booking_types as $key)
   {
-    $params['options'][$key] = get_type_vocab($key);
+    if($is_admin || !in_array($key, $restricted_booking_types))
+       $params['options'][$key] = get_type_vocab($key);
   }
   
   generate_select($params);
diff -Naru mrbs.OLD/functions.inc mrbs/functions.inc
--- mrbs.OLD/functions.inc	2018-02-11 11:49:59.000000000 +0100
+++ mrbs/functions.inc	2018-10-22 09:28:11.913192234 +0200
@@ -1699,13 +1699,10 @@
 {
   global $strftime_format, $year_range;
   
-  // Make sure we've got a date
-  if (empty($day) or empty($month) or empty($year))
-  {
-    $day   = date("d");
-    $month = date("m");
-    $year  = date("Y");
-  }
+  $date = get_day($month, $day, $year);
+  $day   = date("d", $date);
+  $month = date("m", $date);
+  $year  = date("Y", $date);
 
   // Cast the dates to ints to remove any leading zeroes (otherwise
   // JavaScript will treat them as octal numbers)
@@ -2868,6 +2865,14 @@
     $privacy_status = FALSE;
   }
 
+  // ADDITION BY CHRISTOPHER WILKINSON FOR ALL SAINTS CHURCH (2016)
+  global $private_override_by_auth;
+
+  if(strlen(getUserName()) > 0 && isset($private_override_by_auth) && $private_override_by_auth ) {
+    $privacy_status = FALSE;
+  }
+  // END OF BY CHRISTOPHER WILKINSON FOR ALL SAINTS CHURCH (2016)
+
   return $privacy_status;
 }
 
@@ -2999,3 +3004,44 @@
   
   return '';
 }
+
+
+function get_day($month, $day, $year, $incr = 0) {
+
+  global $hidden_days;
+
+  // If we don't know the right date then use today's date
+  if (empty($day) or empty($month) or empty($year))
+  {
+    $day   = date("d");
+    $month = date("m");
+    $year  = date("Y");
+  }
+  else
+  {
+    // Make the date valid if day is more than number of days in month:
+    while (!checkdate($month, $day, $year))
+    {
+      $day--;
+      if ($day <= 0)
+      {
+        $day   = date("d");
+        $month = date("m");
+        $year  = date("Y");   
+        break;
+      }
+    }
+  }
+
+  $date = mktime(12, 0, 0, $month, $day, $year);
+  $date = strtotime($incr." day", $date);
+
+  # Evitamos que el día sea oculto.
+
+  if($incr === 0) $incr = 1;
+  while(is_hidden_day(date("w", $date)) && count($hidden_days) < 7) {
+    $date = strtotime($incr." day", $date);
+  }
+
+  return $date;
+}
diff -Naru mrbs.OLD/lang/lang.es mrbs/lang/lang.es
--- mrbs.OLD/lang/lang.es	2018-02-11 11:49:59.000000000 +0100
+++ mrbs/lang/lang.es	2018-10-19 09:50:45.780405461 +0200
@@ -89,7 +89,7 @@
 $vocab["rep_end_date"]       = "Fecha Tope Repetición";
 $vocab["rep_rep_day"]        = "Día Repetición";
 $vocab["rep_freq"]           = "Frecuencia";
-$vocab["rep_num_weeks"]      = "Número de Semanas";
+$vocab["rep_num_weeks"]      = "Repetición cada";
 $vocab["month_absolute"]     = "En el día";
 $vocab["month_relative"]     = "En el";
 $vocab["skip_conflicts"]     = "Omitir conflictos pasados";
diff -Naru mrbs.OLD/standard_vars.inc.php mrbs/standard_vars.inc.php
--- mrbs.OLD/standard_vars.inc.php	2018-02-11 11:49:59.000000000 +0100
+++ mrbs/standard_vars.inc.php	2018-10-22 09:13:59.293889802 +0200
@@ -24,25 +24,7 @@
 // Get the settings (resolution, etc.) for this area
 get_area_settings($area);
 
-// If we don't know the right date then use today's date
-if (empty($day) or empty($month) or empty($year))
-{
-  $day   = date("d");
-  $month = date("m");
-  $year  = date("Y");
-}
-else
-{
-  // Make the date valid if day is more than number of days in month:
-  while (!checkdate($month, $day, $year))
-  {
-    $day--;
-    if ($day <= 0)
-    {
-      $day   = date("d");
-      $month = date("m");
-      $year  = date("Y");   
-      break;
-    }
-  }
-}
+$date = get_day($month, $day, $year);
+$day   = date("d", $date);
+$month = date("m", $date);
+$year  = date("Y", $date);
